File: Dopamine.InformationModule/ViewModels/InformationViewModel.vb
Imports Microsoft.Practices.Prism.Mvvm
Imports Microsoft.Practices.Prism.Regions
Imports Microsoft.Practices.Prism.Commands
Imports Dopamine.Core.Prism

Namespace ViewModels
    Public Class InformationViewModel
        Inherits BindableBase

#Region "Variables"
        Private ReadOnly mRegionManager As IRegionManager
        Private mPreviousIndex As Integer = 0
        Private mSlideInFrom As Integer
#End Region

#Region "Properties"
        Public NavigateBetweenInformationCommand As DelegateCommand(Of String)

        Public Property SlideInFrom() As Integer
            Get
                Return mSlideInFrom
            End Get
            Set(ByVal value As Integer)
                SetProperty(Of Integer)(mSlideInFrom, value)
            End Set
        End Property
#End Region

#Region "Construction"
        Public Sub New(iRegionManager As IRegionManager)
            Me.mRegionManager = iRegionManager
            Me.NavigateBetweenInformationCommand = New DelegateCommand(Of String)(AddressOf NavigateBetweenInformation)
            ApplicationCommands.NavigateBetweenInformationCommand.RegisterCommand(Me.NavigateBetweenInformationCommand)
            Me.SlideInFrom = 30
        End Sub
#End Region

#Region "Functions"
        Private Sub NavigateBetweenInformation(iIndex As String)

            If String.IsNullOrWhiteSpace(iIndex) Then Return

            Dim index As Integer = 0

            Integer.TryParse(iIndex, index)

            If index = 0 Then Return

            Me.SlideInFrom = If(index <= mPreviousIndex, -30, 30)

            mPreviousIndex = index

            mRegionManager.RequestNavigate(RegionNames.InformationRegion, Me.GetPageForIndex(index))
        End Sub

        Private Function GetPageForIndex(iIndex As Integer) As String

            Dim page As String = String.Empty

            Select Case iIndex
                Case 1
                    page = GetType(Views.InformationHelp).FullName
                Case 2
                    page = GetType(Views.InformationAbout).FullName
                Case Else
                    page = GetType(Views.InformationHelp).FullName
            End Select

            Return page
        End Function
#End Region
    End Class
End Namespace

File: Dopamine.InformationModule/ViewModels/InformationViewModel.cs
using Dopamine.Core.Prism;
using Microsoft.Practices.Prism.Commands;
using Microsoft.Practices.Prism.Mvvm;
using Microsoft.Practices.Prism.Regions;

namespace Dopamine.InformationModule.ViewModels
{
    public class InformationViewModel : BindableBase
    {
        #region Variables
        private readonly IRegionManager regionManager;
        private int previousIndex = 0;
        private int slideInFrom;
        #endregion

        #region Commands    
        public DelegateCommand<string> NavigateBetweenInformationCommand;
        #endregion

        #region Properties
        public int SlideInFrom
        {
            get { return this.slideInFrom; }
            set { SetProperty<int>(ref this.slideInFrom, value); }
        }
        #endregion
    
        #region Construction
        public InformationViewModel(IRegionManager regionManager)
        {
            this.regionManager = regionManager;
            this.NavigateBetweenInformationCommand = new DelegateCommand<string>(NavigateBetweenInformation);
            ApplicationCommands.NavigateBetweenInformationCommand.RegisterCommand(this.NavigateBetweenInformationCommand);
            this.SlideInFrom = 30;
        }
        #endregion

        #region Private
        private void NavigateBetweenInformation(string iIndex)
        {
            if (string.IsNullOrWhiteSpace(iIndex))
                return;

            int index = 0;

            int.TryParse(iIndex, out index);

            if (index == 0)
                return;

            this.SlideInFrom = index <= this.previousIndex ? -30 : 30;

            this.previousIndex = index;

            this.regionManager.RequestNavigate(RegionNames.InformationRegion, this.GetPageForIndex(index));
        }

        private string GetPageForIndex(int iIndex)
        {

            string page = string.Empty;

            switch (iIndex)
            {
                case 1:
                    page = typeof(Views.InformationHelp).FullName;
                    break;
                case 2:
                    page = typeof(Views.InformationAbout).FullName;
                    break;
                default:
                    page = typeof(Views.InformationHelp).FullName;
                    break;
            }

            return page;
        }
        #endregion
    }
}
